core/flow/rule_manager.go

```diff
@@ -513,5 +546,28 @@ func IsValidRule(rule *Rule) error {
 	if rule.StatIntervalInMs > 10*60*1000 {
 		logging.Info("StatIntervalInMs is great than 10 minutes, less than 10 minutes is recommended.")
 	}
+	if rule.TokenCalculateStrategy == AdaptiveMemory {
+		if rule.RiskThreshold >= rule.SafeThreshold {
+			return errors.New("rule.RiskThreshold should be less than rule.SafeThreshold")
+		}
+
+		if rule.LowWaterMark == 0 {
```

Comment:
Maybe we should check HighWaterMark , LowWaterMark, RiskThreshold, SafeThreshold is not negative.

Reply:
updated